src/Razor/Microsoft.AspNetCore.Razor.Language/src/Legacy/SeekableTextReader.cs

```diff
@@ -8,9 +8,10 @@
 
 namespace Microsoft.AspNetCore.Razor.Language.Legacy
 {
-    internal class SeekableTextReader : TextReader, ITextDocument
+    internal sealed class SeekableTextReader : TextReader, ITextDocument
     {
         private readonly RazorSourceDocument _sourceDocument;
+        private readonly string _filePath;
```

Comment:
?

Reply:
Oops, messed up the PR. `source.get_FilePath` shows on the profiler (as do some of the other property accessors) as part of thew `UpdateState` function call which appears very frequently. The FilePath does not change for a source document, so it seemed like a easy enough fix.